### PR TITLE
Update GitHub Actions handling artefacts to @v4

### PR DESCRIPTION
**Pull request by probonopd**

### CODE REVIEW

**Comment by probonopd** [comment]
@TheAssassin please advise. This type of stuff is why I don't like Docker. Containers seem to just disappear...

**Comment by TheAssassin** [comment]
It's quite apparent that this is not a container issue.

File: ci/build-in-docker.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ case "$ARCH" in
 esac
 
 # libassuan-static is supported only from 3.19 onwards
-image="$image_prefix"/alpine:3.19
+image="$image_prefix"/alpine:latest
```
Comment (TheAssassin) [inline ci/build-in-docker.sh:34]: We want to use a specific version.

Comment (TheAssassin) [reply ci/build-in-docker.sh:34]: `image_prefix` likely isn't needed any more, specifying the platform is sufficient nowadays and has been for a few years. Please try removing that variable altogether.

Comment (probonopd) [reply ci/build-in-docker.sh:34]: Indeed. 👍 
I don't know why it fails for ARM now.

**Review by TheAssassin — APPROVED** [review]

**Comment by TheAssassin** [comment]
Next time please clean up your history _before_ switching on some automerge.

**Pull request merged**